x-pack/plugins/security_solution/server/lib/timeline/routes/notes/get_notes.ts

```diff
@@ -107,6 +115,64 @@ export const getNotesRoute = (router: SecuritySolutionPluginRouter) => {
               sortOrder,
               filter,
             };
+
+            const associatedFilter = queryParams?.associatedFilter;
```

Comment:
I think it is a lot of new code and a good opportunity to add tests. 

See if it is possible to add a proper API integration tests here : `x-pack/test/security_solution_api_integration`

Reply:
I already have plans of adding more tests, but because I wasn't sure this was the right way of doing this filters, I didn't want to waste time writing test on a code I was unsure of... I also have a follow up PR that needs to get in before `8.16` so I will spend time writing tests after feature freeze

Reply:
Makes sense.

Reply:
I changed my mind and added some api integration tests in [this commit](https://github.com/elastic/kibana/pull/195501/commits/8b2ae641cedf7ba02bdb1d6dcbe01eae74be9045). These cover only the added code, I have a separate [PR](https://github.com/elastic/kibana/pull/196225) that adds api integration tests for the rest of the logic within the `getNotes` endpoint, because I realized while adding tests for this that the logic was actually wrong! It didn't make sense to fix the wrong existing logic in this PR that adds new logic...